src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -359,6 +360,92 @@ public void validateNoKeyPersonnelDuplicates(DataAccessRequestData darData) {
     }
   }
 
+  @VisibleForTesting
+  public void validatePersonnelInSameInstitution(User user, DataAccessRequestData darData) {
+    Institution submitterInstitution = user.getInstitution();
+    String piEmail = darData.getPiEmail();
+    String soEmail = darData.getSigningOfficialEmail();
+    String itEmail = darData.getItDirectorEmail();
+    List<String> collaboratorsEmails =
+        darData.getInternalCollaborators().stream().map(Collaborator::getEmail).toList();
+    List<String> labStaffEmails =
+        darData.getLabCollaborators().stream().map(Collaborator::getEmail).toList();
+
+    List<String> invalidMembers = new ArrayList<>();
+
+    if (emailDoesNotMatchInstitution(submitterInstitution, piEmail)) {
+      invalidMembers.add("Principal Investigator: " + piEmail);
+    }
+    if (emailDoesNotMatchInstitution(submitterInstitution, soEmail)) {
+      invalidMembers.add("Signing Official: " + soEmail);
+    }
+
+    if (emailDoesNotMatchInstitution(submitterInstitution, itEmail)) {
+      invalidMembers.add("IT Director: " + itEmail);
+    }
+
+    getErrorSummary(
+            collaboratorsEmails,
+            submitterInstitution,
+            "Internal Collaborator member: ",
+            "Internal Collaborator members: ")
+        .ifPresent(invalidMembers::add);
+
+    getErrorSummary(
+            labStaffEmails, submitterInstitution, "Lab staff member: ", "Lab staff members: ")
+        .ifPresent(invalidMembers::add);
+
+    if (!invalidMembers.isEmpty()) {
+      throw new IllegalArgumentException(
+          "All listed personnel must share the same institutional affiliation.  The following list of roles and members must have email addresses associated with your institution: "
+              + String.join(", ", invalidMembers));
+    }
+  }
+
+  private Optional<String> getErrorSummary(
+      List<String> emails,
+      Institution institution,
+      String categorySingular,
+      String categoryPlural) {
+    List<String> errors = findEmailAddressesNotInInstitution(emails, institution);
+    if (!errors.isEmpty()) {
+      return Optional.of(buildSingleErrorFromErrorList(errors, categorySingular, categoryPlural));
+    }
+    return Optional.empty();
+  }
+
+  private List<String> findEmailAddressesNotInInstitution(
+      List<String> emailAddresses, Institution institution) {
+    ArrayList<String> errors = new ArrayList<>();
+    emailAddresses.forEach(
+        collaborator -> {
+          if (emailDoesNotMatchInstitution(institution, collaborator)) {
+            errors.add(collaborator);
+          }
+        });
+    return errors;
```

Comment:
Since you're using a `List<String>` to produce another `List<String>`, using a streaming operation is usually a better way to express this.